lighthouse-core/lib/url-shim.js

```diff
@@ -172,12 +179,34 @@ class URLShim extends URL {
       return false;
     }
   }
+
+  /**
+   * @param {string} hostname Either a `new URL(url).hostname` or a `networkRequest.parsedUrl.host`
+   * @return {boolean}
+   */
+  static isLikeLocalhost(hostname) {
```

Comment:
Let's add `.local` tld too.

Reply:
A few others too, maybe... https://serverfault.com/questions/1006000/is-tld-local-not-a-local-tld-anymore

Reply:
Looking into this... our `isLikeLocalhost` is used by `isSecureRequest()` which is used by `is-on-https`. So we're just not reporting localhosty things as failures for HTTPS.

In chromium there's a few policies around this:
- http://www.chromium.org/Home/chromium-security/deprecating-powerful-features-on-insecure-origins
- https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features
- http://www.chromium.org/Home/chromium-security/marking-http-as-non-secure

as i understand it, there's a small semantic split between Lock-Icon-secure and this-secure-context-can-have-powerful-features.. for example... no lock icon and popup says "connection is not secure" but at the same time it's a "secure context" as powerful getUserMedia is allowed:

![image](https://user-images.githubusercontent.com/39191/109071225-4b9b8a00-76a8-11eb-9098-0d8c8acb627a.png)

does this distinction matter to us and is-on-https? probably not, but i wanted to call it out because we could misuse isLikeLocalhost i imagine…

---------

i'm a little wary to put all the domains mentioned in that SO in here, so i'm doublechecking against spec language... basically these two guys:

* https://w3c.github.io/webappsec-secure-contexts/#localhost
* https://tools.ietf.org/html/draft-ietf-dnsop-let-localhost-be-localhost

The chromium impl i find is https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/platform/weborigin/security_origin.cc;l=524-529;drc=61219381be5d2405dcd1aac31fc60eaf6342896a 

and exploring we find https://bugs.chromium.org/p/chromium/issues/detail?id=1153336 which says these things don't always agree, but IMO it's in the weeds that we don't care about.


------

I'm gonna allow any host ending in `.localhost` as that's what the ietf rfc has. And i'll add a comment  about the remaining edgecases with loopback/etc.